pkgs/development/libraries/jemalloc/default.nix

```
@@ -47,10 +47,15 @@ stdenv.mkDerivation rec {
     ++ lib.optional (stdenv.isDarwin && stdenv.isx86_64) "--with-lg-vaddr=48"
   ;
 
-  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-Wno-error=array-bounds";
+  env = {
+    NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-Wno-error=array-bounds";
+    # Tries to reference undefined shlibs
+    NIX_LDFLAGS = lib.optionalString (stdenv.cc.bintools.isLLVM && lib.versionAtLeast stdenv.cc.bintools.version "17") "--allow-shlib-undefined";
+  };
 
-  # Tries to link test binaries binaries dynamically and fails
-  doCheck = !stdenv.hostPlatform.isStatic;
+  # static: Tries to link test binaries binaries dynamically and fails
+  # llvm: Tries linking with an undefined symbol and fails.
+  doCheck = !stdenv.hostPlatform.isStatic && !(stdenv.cc.bintools.isLLVM && lib.versionAtLeast stdenv.cc.bintools.version "17");
```

```suggestion
  doCheck = !stdenv.hostPlatform.isStatic;
  checkFlags = lib.optionals condition [ "LDFLAGS=-lc++" ];
```

fixes the tests for me. it seems that this is broken prior to llvm17 too.

---

tho maybe
```nix
buildFlags = lib.optionals condition [ "LDFLAGS=-lc++" ];
```
is better as that will link jemalloc.so to libc++, which is similar to when compiled with gcc which links to `libstdc++`

---

Yes, that fixed it.